lib/waterline-schema/schema.js

```diff
@@ -208,6 +208,13 @@ module.exports = function schemaBuilder(collections) {
         throw new Error('The attribute `' + attributeName + '` on the `' + collection.identity + '` model is a plural association and can\'t have a required flag set. Plural associations can only be populated.');
       }
 
+      // Ensure any plural associations don't point to themselves
```

Comment:
I think we need to narrow this check to also check that the model identities are the same, since same-name `via` is only forbidden for reflexive associations.  For example, the following ontology is valid:


```javascript
// in `attributes` of api/models/User.js
friends: { collection: 'Pet', via: 'friends' },
```

```javascript
// in `attributes` of api/models/Pet.js
friends: { collection: 'User', via: 'friends' },
```